File: Problems/876_Middle_of_the_Linked_List.py
# Definition for singly-linked list.
class ListNode:
    def __init__(self, val=0, next=None):
        self.val = val
        self.next = next
from typing import Optional
class Solution:
    def middleNode(self, head: Optional[ListNode]) -> Optional[ListNode]:
        slow = head
        fast = head
        count = 0
        while fast:
            if count %2:
                slow = slow.next
            fast = fast.next
            count += 1
        return slow

File: Problems/876_Middle_of_the_Linked_List.c

// Definition for singly-linked list.
struct ListNode {
    int val;
    struct ListNode *next;
};

struct ListNode* middleNode(struct ListNode* head){
    struct ListNode* curr = head;
    struct ListNode* middle = head;
    int cnt = -1;
    while (curr){
        curr = curr -> next;
        if (cnt %2 == 0){
            middle = middle -> next;
        }
        cnt += 1;
    }
    return middle;
}